component/otelcol/receiver/loki/loki.go

```diff
@@ -106,34 +106,30 @@ func (c *Component) Update(newConfig component.Arguments) error {
 	return nil
 }
 
-// parsePromtailEntry creates new stanza.Entry from promtail entry
-func parsePromtailEntry(inputEntry loki.Entry) *entry.Entry {
-	outputEntry := entry.New()
-	outputEntry.Body = inputEntry.Entry.Line
-	outputEntry.Timestamp = inputEntry.Entry.Timestamp
+// Create a new Otlp Logs entry from a Promtail entry
+func convertLokiEntryToPlog(lokiEntry loki.Entry) plog.Logs {
+	logs := plog.NewLogs()
+
+	lr := logs.ResourceLogs().AppendEmpty().ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
+
+	if filename, exists := lokiEntry.Labels["filename"]; exists {
+		filenameStr := string(filename)
+		// The `promtailreceiver` from the opentelemetry-collector-contrib
+		// repo adds these two labels based on these "semantic conventions
+		// for log media".
+		// https://opentelemetry.io/docs/reference/specification/logs/semantic_conventions/media/
+		// We're keeping them as well, but we're also adding the `filename`
+		// attribute so that it can be used from the
+		// `loki.attribute.labels` hint for when the opposite OTel -> Loki
+		// transformation happens.
+		lr.Attributes().PutStr("log.file.path", filenameStr)
+		lr.Attributes().PutStr("log.file.name", path.Base(filenameStr))
+	}
```

Comment:
The `promtailreceiver` does no longer exist, and the `lokireceiver` replacement does not follow the same logic (maybe because it's considered more of a network-based source?), and these conventions are experimental anyway.

I'd recommend we remove this to keep us more in-line with upstream and adding a comment like this
```
// TODO: We should consult with upstream to add the `log.file.path` and `log.file.name` attributes once the semantic conventions for log media are mature 
// https://opentelemetry.io/docs/reference/specification/logs/semantic_conventions/media/
```

Reply:
Ok, I'll make the change separately. If you'd put a `TODO(@tpaschalis)` comment I'd appreciate it

Reply:
@ptodev Can you add the TODO comment here prior to merging?

Reply:
Yes, just added it 👍